frontend/src/views/WheresMyGeneV2/components/GeneSearchBar/components/ShareButton/utils.tsx

```diff
@@ -87,20 +87,105 @@ export const loadStateFromQueryParams = ({
 }): LoadStateFromURLPayload | null => {
   if (isSSR()) return null;
 
-  const paramsToRemove = [];
+  const paramsToRemove: string[] = [];
 
   // Checks if the URL has any query params related to shared filter state
   // If so, it will update the state with the values from the URL
 
-  // Check for version
+  const { delimiter } = getMetadataForShareLinkParsing({
+    params,
+    paramsToRemove,
+  });
+
+  // Check for filter properties
+  const newSelectedFilters = getNewSelectedFilters({
```

Comment:
extract selectedFilters parsing in its own function to reduce cognitive complexity